include/examples/simple_behavior.h

```diff
@@ -0,0 +1,192 @@
+   *          update. Use this to define the primary content of your behavior,
+   *          e.g., to update and command new joint torques.  Additionally, this
+   *          code can monitor if the behavior has terminated and update flags
+   *          appropriately. In this example, this function reads the updated
+   *          joint velocities and maintains state machine corresponding to the
+   *          member enum `state_`.
+   */
+  void Update() override
+  {
+    // Update joint velocities from robot
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    switch (state_)
+    {
+      case RUNNING:  // in a running state
+        // Set joint torques to running levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), RUN_TORQUE);
+        break;
+
+      case STOPPING:  // in a stopping state
+        // Lower the joint torques to slowing levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), SLOW_TORQUE);
+        break;
+    }
+
+    // Set joint torques
+    robot_->SetTorques(desired_torques_);
+  }
+
+  /**
+   * @brief Stops the behavior
+   * @details This code is run when a behavior is directed to stop running by
+   *          the behavior manager. This should trigger a shutdown process for
+   *          the behavior, e.g. slowing the motor speed. The upcoming behavior
+   *          is passed as an argument to permit different shutdown procedures
+   *          for different on-deck behaviors. In this example, the internal
+   *          state is modified to indicate the behavior should stop.
+   * @param next_behavior the upcoming behavior
+   */
+  void Stop(const kodlab::Behavior<SimpleRobot> &next_behavior) override
+  {
+    // Indicate that behavior should stop
+    state_ = STOPPING;
+    std::fprintf(stdout, "Stopping %s\n", get_name().c_str());
+  }
+
+  /**
+   * @brief Returns the active state of the behavior
+   * @details This function identifies if the behavior is active. By default, it
+   *          simply accesses the `active_` variable. More advanced
+   *          implementations may use robot state to make this determination.
+   * @return true if behavior is running, false otherwise
+   */
+  bool Running() override { return active_; }
+
+  /**
+   * @brief Indicates if behavior is finished and ready to transition
+   * @details This function indicates to the behavior manager whether the
+   *          behavior is prepared to terminate and transition to the next
+   *          behavior. In its simplest form, this function simply accesses
+   *          the internal behavior active flag. This function can be used as a
+   *          mode switch guard.
+   * @param next_behavior the upcoming behavior
+   * @return true if behavior is ready to transition, false otherwise
+   */
+  bool ReadyToSwitch(const kodlab::Behavior<SimpleRobot> &next_behavior) override
```

Comment:
how might a behavior use this next_behavior to make decisions in its current form? 
Would it require a dynamic_cast? 
Or would it only be possible by name?

Reply:
In the present form, the user can either access the `Behavior`'s name and use this for selective switching conditions, or attempt to downcast and use the full behavior.  The former is the intended usage, using only default `Behavior` members for making determinations about the incoming behavior.  However, the latter is permitted.

Reply:
For the time being, this will remain as is, and it is up to the user to handle downcasting if desired.  In the future, we could implement a `get_type` method that returns the behavior type by adapting a CRTP structure for `Behavior`.